numpy/_core/src/multiarray/stringdtype/static_string.c

```diff
@@ -698,7 +661,7 @@ NpyString_dup(const npy_packed_static_string *in,
         return -1;
     }
     int used_malloc = 0;
-    if (in_allocator == out_allocator && is_a_vstring(in)) {
+    if (in_allocator == out_allocator && !is_short_string(in)) {
```

Comment:
Cannot be missing here, so this substitution is fine (and removes the only use of `is_a_vstring`).